nilearn/datasets/atlas.py

```diff
@@ -320,104 +324,290 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
-            label_file = 'HarvardOxford-Cortical-Lateralized.xml'
-            lateralized = True
+    if atlas_source == 'HarvardOxford':
+        if atlas_name[0] == 'c':
+            if ('cort-maxprob' in atlas_name and symmetric_split
+                    or 'cortl-maxprob' in atlas_name):
+                split_name = atlas_name.split('-', 1)
+                atlas_name = 'cortl-{}'.format(split_name[1])
+                label_file = 'HarvardOxford-Cortical-Lateralized.xml'
+                lateralized = True
+            else:
+                label_file = 'HarvardOxford-Cortical.xml'
+                lateralized = False
         else:
-            label_file = 'HarvardOxford-Cortical.xml'
+            label_file = 'HarvardOxford-Subcortical.xml'
             lateralized = False
     else:
-        label_file = 'HarvardOxford-Subcortical.xml'
+        label_file = "Juelich.xml"
         lateralized = False
     label_file = os.path.join(root, label_file)
-
-    atlas_file = os.path.join(root, 'HarvardOxford',
-                              'HarvardOxford-' + atlas_name + '.nii.gz')
-
+    atlas_file = os.path.join(root, atlas_source,
+                              '{}-{}.nii.gz'.format(atlas_source,
+                                                    atlas_name))
     atlas_img, label_file = _fetch_files(
         data_dir,
-        [(atlas_file, url, opts), (label_file, url, opts)],
+        [(atlas_file, url, opts),
+         (label_file, url, opts)],
         resume=resume, verbose=verbose)
-
     names = {}
     from xml.etree import ElementTree
     names[0] = 'Background'
     for label in ElementTree.parse(label_file).findall('.//label'):
         names[int(label.get('index')) + 1] = label.text
     names = list(names.values())
+    return atlas_img, names, lateralized
 
-    if not symmetric_split:
-        return Bunch(maps=atlas_img, labels=names)
-
-    if atlas_name in ("cort-prob-1mm", "cort-prob-2mm",
-                      "sub-prob-1mm", "sub-prob-2mm"):
-        raise ValueError("Region splitting not supported for probabilistic "
-                         "atlases")
 
-    atlas_img = check_niimg(atlas_img)
-    if lateralized:
-        return Bunch(maps=atlas_img, labels=names)
+def _merge_probabilistic_maps_juelich(atlas_data, names):
+    """Helper function for fetch_atlas_juelich.
+    This function handles probabilistic juelich atlases
+    when symmetric_split=False. In this situation, we need
+    to merge labels and maps corresponding to left and right
+    regions.
+    """
+    if atlas_data.ndim != 4:
+        raise ValueError("Expected a 4D atlas here.")
+    new_names = np.unique([name[:-2] if name.endswith(' L')
+                           or name.endswith(' R') else name for name in names])
+    new_atlas_data = np.zeros((*atlas_data.shape[:3],
+                               len(new_names) - 1))
+    for i, name in enumerate(new_names):
+        if name != "Background":
```

Comment:
I think it is the fact that atlas_data has the last dimension 121 but the names has a length of 122. Therefore, we excluded the Background while merging the maps while keeping it in the new names.

Reply:
@achamma723 we need to keep the "-1" in the `new_atlas_data` last dimension because we don't want to build a map for the background (see my last commit). @jeromedockes is this what you were referring to?

Reply:
yes it was -- why no label and no map for the background. we could maybe add a short comment explaining why it is not included and hence the `-1`?

Reply:
@jeromedockes maybe I'm missing something, but if you do (on master):

```python
from nilearn import datasets
from nilearn import image
d = datasets.fetch_atlas_harvard_oxford("sub-prob-1mm")
print(len(d.labels))
print(image.get_data(d.maps).shape)
```

You get:

```
22
(182, 218, 182, 21)
```

So my conclusion was that we don't return a map for the background (but the label does exist).
Does this make sense?